core/templates/dev/head/components/forms/schema_editors/SchemaBasedHtmlEditorDirective.js

```diff
@@ -28,6 +30,200 @@ oppia.directive('schemaBasedHtmlEditor', [
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/components/forms/schema_editors/' +
         'schema_based_html_editor_directive.html'),
-      restrict: 'E'
+      restrict: 'E',
+      link: function($scope, element, attrs) {
+        // A mutation observer to detect changes in the RTE Editor
+        var observer = new MutationObserver(function(mutations) {
+          // A condition to detect if mutation involves addition of node by
+          // drag and drop method.
+          if (mutations[mutations.length - 1].addedNodes[0]) {
+            // This condition checks if added node is image type.
+            if (mutations[mutations.length - 1]
+              .addedNodes[0].nodeName === 'IMG'){
+              // Gets the added image node
+              var addedImgNode = mutations[mutations.length - 1].addedNodes[0];
+              addedImgNode.classList.add('oppia-noninteractive-image');
+              addedImgNode.classList.add('block-element');
+
+              var MODE_EMPTY = 1;
+              var MODE_SAVED = 3;
+
+              var SOURCE_SEPARATOR = 21;
+              var FILEPATH_SAPERATOR = 48;
+
+              var OUTPUT_IMAGE_FORMAT = 'png';
+              var OUTPUT_IMAGE_MAX_WIDTH_PX = 490;
+
+              $scope.data = {mode: MODE_EMPTY, metadata: {}};
```

Comment:
I would rename this to something more meaningful. What is the purpose of this data for? Should it be called imageData?